EventFilter/OnlineMetaDataRawToDigi/plugins/OnlineMetaDataRawToDigi.cc

```diff
@@ -0,0 +1,145 @@
+// -*- C++ -*-
+//
+// Package:    EventFilter/OnlineMetaDataRawToDigi
+// Class:      OnlineMetaDataRawToDigi
+//
+/**\class OnlineMetaDataRawToDigi OnlineMetaDataRawToDigi.cc EventFilter/OnlineMetaDataRawToDigi/plugins/OnlineMetaDataRawToDigi.cc
+
+ Description:  Producer to unpack event meta-data from soft-FED 1022
+
+*/
+//
+// Original Author:  Remigius K Mommsen (Fermilab)
+//         Created:  Wed, 22 Nov 2017 18:19:56 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+#include "DataFormats/FEDRawData/interface/FEDHeader.h"
+
+#include "DataFormats/OnlineMetaData/interface/DCSRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineLuminosityRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineMetaDataRaw.h"
+
+
+//
+// class declaration
+//
+
+
+class OnlineMetaDataRawToDigi : public edm::stream::EDProducer<> {
+public:
+  explicit OnlineMetaDataRawToDigi(const edm::ParameterSet&);
+  ~OnlineMetaDataRawToDigi() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  reco::BeamSpot getBeamSpot(const online::BeamSpot_v1&) const;
+
+  edm::EDGetTokenT<FEDRawDataCollection> dataToken_;
+};
+
+OnlineMetaDataRawToDigi::OnlineMetaDataRawToDigi(const edm::ParameterSet& iConfig)
+{
+  edm::InputTag dataLabel = iConfig.getParameter<edm::InputTag>("onlineMetaDataInputLabel");
+  dataToken_=consumes<FEDRawDataCollection>(dataLabel);
+
+  produces<DCSRecord>("dcsRecord");
+  produces<OnlineLuminosityRecord>("onlineLuminosityRecord");
+  produces<reco::BeamSpot>("onlineBeamSpot");
```

Comment:
Please drop the product instance strings, unless there is a good reason to have  e.g. 
```OnlineLuminosityRecord_onlineMetaDataRawToDigi_onlineLuminosityRecord_PROCESS``` in the branch name instead of simply ```OnlineLuminosityRecord_onlineMetaDataRawToDigi__PROCESS```